### DISCUSSION TITLE
Using a third-party library to inherit component themes.

### DISCUSSION THREAD

**Discussion by TareqRafed** [chakra-ui/chakra-ui#8447] [Q&A] [answered]
I'm facing an issue when I want to get this ``<CurrencyInputBase />`` inherit `themes.components.Input` theme
I tried `useStyleConfig` but didn't work. 

```ts
const CurrencyInputBase = chakra(CurrencyInput); // CurrencyInput is a controlled <input> element
const CurrencyField = ({ variant, ...rest }: CurrencyInputProps & InputProps) => {
  const styles = useStyleConfig('Input', { variant }) // gets computed but doesn't apply styles
  return (
    <CurrencyInputBase 
      __css={styles}
      {...rest}
    />
  );
};
```
I also tried `as` keyword, it breaks functionality of the input, I'm using **v2.4.2**

**Comment by veloware** [comment]
Its probably that your `__css` prop isn't making it to the underlying input element, or it is being overridden by another `__css` prop inside your `CurrencyInput`.

Failing that, you could spread your styles?

```
<CurrencyInputBase 
      {...styles}
      {...rest}
    />
```

**Comment by segunadebayo** [comment, marked as answer]
`useStyleConfig` is v2. Wrap the third-party input with `chakra(CurrencyInput)` and pass the input recipe.
